src/util/Conversions.h

```diff
@@ -568,10 +571,37 @@ inline string removeLeadingZeros(const string& orig) {
 }
 
 // _____________________________________________________________________________
-bool isXsdValue(const string val) {
+bool isXsdValue(const string& val) {
   // starting the search for "^^ at position 1 makes sure it's not in the
   // quotes as we already checked that the first char is the first quote
-  return val.size() > 0 && val[0] == '\"' &&
-         val.find("\"^^", 1) != string::npos;
+  return !val.empty() && val[0] == '\"' && val.find("\"^^", 1) != string::npos;
+}
+
+// _____________________________________________________________________________
+bool isNumeric(const string& val) {
+  if (val.empty()) {
+    return false;
+  }
+  size_t start = (val[0] == '-') ? 1 : 0;
+  size_t posNonDigit = val.find_first_not_of("0123456789", start);
+  if (posNonDigit == string::npos) {
+    return true;
+  }
+  if (val[posNonDigit] == '.') {
+    return posNonDigit + 1 < val.size() &&
+           val.find_first_not_of("0123456789", posNonDigit + 1) == string::npos;
+  }
+  return false;
+}
+
+// _____________________________________________________________________________
+string convertNumericToIndexWord(const string& val) {
+  bool wasInt = false;
+  string tmp = val;
+  if (tmp.find('.') == string::npos) {
+    tmp += ".0";
+    wasInt = true;
+  }
```

Comment:
Is the "numeric" type also handled correctly during Index build? Does the NTriples parser convert such literals to strings or do we also have to add the check for `isNumeric` in Index.cpp?

Reply:
I was under the impression that this is not valid in NTriple files but honestly I'm not sure.

Reply:
Seems like that impression was true https://www.w3.org/TR/rdf11-concepts/#dfn-literal

Reply:
I am sorry, you are right. NTriples is indeed a simple format